src/MyEventEmitter.js

```diff
@@ -1,12 +1,122 @@
 'use strict';
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+class MyEventEmitter {
+  constructor() {
+    this.events = {};
+  }
+
+  validateEventName(eventName) {
+    if (typeof eventName !== 'string') {
+      throw new Error('The event name supposed to be a string');
+    }
+  }
+
+  validateListener(listener) {
+    if (typeof listener !== 'function') {
+      throw new Error('The listener supposed to be a fanction');
+    }
+  }
+
+  on(eventName, listener) {
+    this.validateEventName(eventName);
+    this.validateListener(listener);
+
+    if (!(eventName in this.events)) {
+      this.events[eventName] = [listener];
+    } else {
+      this.events[eventName].push(listener);
+    }
+
+    return this;
+  }
+
+  once(eventName, listener) {
+    this.validateEventName(eventName);
+    this.validateListener(listener);
+
+    if (!this.events[eventName]) {
+      this.events[eventName] = [];
+    }
+
+    const onceFn = (...args) => {
+      listener(args);
```

Comment:
The listener should be invoked with individual arguments, not an array of arguments. Change 'listener(args)' to 'listener(...args)'.